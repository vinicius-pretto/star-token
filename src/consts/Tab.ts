enum Tab {
  TOKENS = "TOKENS",
  CSS = "CSS",
  SCSS = "SCSS",
  JSON = "JSON",
}

export default Tab;
