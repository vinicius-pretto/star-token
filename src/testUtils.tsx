import * as React from "react";
import { render } from "@testing-library/react";
import { Provider as ReduxProvider } from "react-redux";
import configureStore, { initialState } from "./ui/redux/store";

const Providers = ({ children, state }) => {
  return (
    <ReduxProvider store={configureStore(state)}>{children}</ReduxProvider>
  );
};

const customRender = (ui, state = initialState) => {
  const options = {
    wrapper: (props) => <Providers {...props} state={state} />,
  };
  return render(ui, options);
};

export * from "@testing-library/react";
export { customRender as render };
