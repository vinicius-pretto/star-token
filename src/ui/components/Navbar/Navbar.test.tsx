import * as React from "react";
import { render } from "../../../testUtils";
import NavBar from "./Navbar";
import Tab from "../../../consts/Tab";

describe("Navbar", () => {
  it("snapshot", () => {
    const { asFragment } = render(
      <NavBar onClick={jest.fn()} tabSelected={Tab.TOKENS} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
