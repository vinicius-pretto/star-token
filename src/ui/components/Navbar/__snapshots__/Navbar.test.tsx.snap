// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<DocumentFragment>
  <nav
    class="border-bottom py-xs px-sm"
  >
    <ul
      class="d-flex flex-row p-0"
    >
      <li>
        <a
          class="section-title"
          href="#"
        >
          Tokens
        </a>
      </li>
      <li>
        <a
          class="label pr-md"
          href="#"
        >
          CSS
        </a>
      </li>
      <li>
        <a
          class="label pr-md"
          href="#"
        >
          SCSS
        </a>
      </li>
      <li>
        <a
          class="label pr-md"
          href="#"
        >
          JSON
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
